### PR TITLE
[bitnami/postgresql-ha] Chart standardised and bump to PSQL@14

### PR DESCRIPTION
**Pull request by FraPazGal**
**Description of the change**

This PR updates the PostgreSQL image's version to the latest major, `v14`, as well as standarizes the templates and values. These changes can be sumarised in the following:

- Added missing parameters and modified templates to follow `bitnami/charts` templating.
- Image parameters that used `imageNameImage` are now under `imageName.image`
- `containerPort` parameters are now found by `containerPorts.xxxx`
- `service.port` parameters are now found by `service.ports.xxxx`
- `imageName.securityContext` parameters are now under `imageName.podSecurityContext`
- `serviceAccount.enabled` parameter has being renamed to `serviceAccount.create`
- `postgresql.updateStrategyType` parameter has been renamed to `postgresql.updateStrategy.type`
- Added the `pgpool.authenticationMethod` parameter. Use "md5" for older PSQL versions and the default value of "scram-sha-256" when using PSQL > v14.

**Benefits**

**Possible drawbacks**

**Checklist**

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/postgresql-ha/values.yaml [hunk 1]
```diff
@@ -215,31 +187,55 @@ postgresql:
...
+  ## ref: https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods
+  ##
+  terminationGracePeriodSeconds: ""
+  ## @param postgresql.updateStrategy.rollingUpdate PostgreSQL statefulset rolling update configuration parameters
+  ## @param postgresql.updateStrategy.type Postgresql statefulset strategy type
+  ## ref: https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#update-strategies
+  ##
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate: {}
```
Comment (fmulero) [inline bitnami/postgresql-ha/values.yaml:212]: if `updateStrategy.type` is set to "OnDelete" the parameter `updateStrategy.rollingUpdate` could cause problems. This approach is better:

https://github.com/bitnami/charts/blob/5af36c98d53a5ad3911e0f4dd7fa5c63bb2b37a9/bitnami/apache/values.yaml#L357-L367

File: bitnami/postgresql-ha/values.yaml [hunk 2]
```diff
@@ -154,9 +129,10 @@ postgresql:
   ## ref: https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#update-strategies
   ##
   updateStrategyType: RollingUpdate
```
Comment (fmulero) [inline bitnami/postgresql-ha/values.yaml:131]: I think it should be like:

https://github.com/bitnami/charts/blob/5af36c98d53a5ad3911e0f4dd7fa5c63bb2b37a9/bitnami/apache/values.yaml#L357-L367

**Review by fmulero — CHANGES_REQUESTED** [review]
Hi @FraPazGal Great PR! Please have a look to my comments

File: bitnami/postgresql-ha/values.yaml (hunk 2 above)
Comment (FraPazGal) [reply bitnami/postgresql-ha/values.yaml]: My bad, this param should have been replaced by the one which you commented on above.

File: bitnami/postgresql-ha/values.yaml [hunk 3]
```diff
@@ -1222,45 +1296,49 @@ metrics:
...
-## @param volumePermissionsImage.pullSecrets Specify docker-registry secret names as an array
-##
-volumePermissionsImage:
-  registry: docker.io
-  repository: bitnami/bitnami-shell
-  tag: 10-debian-10-r387
-  ## Specify a imagePullPolicy. Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
-  ## ref: https://kubernetes.io/docs/user-guide/images/#pre-pulling-images
-  ##
-  pullPolicy: IfNotPresent
```
[the remark this answers is not in the document]
Comment (FraPazGal) [reply bitnami/postgresql-ha/values.yaml:1244]: Good catch! Fixed

File: bitnami/postgresql-ha/values.yaml [hunk 4]
```diff
@@ -44,54 +43,32 @@ global:
...
-## Pod Service Account
-## ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
-##
-serviceAccount:
-  ## @param serviceAccount.enabled Enable service account (Note: Service Account will only be automatically created if `serviceAccount.name` is not set)
-  ##
-  enabled: false
-  ## @param serviceAccount.name Name of an already existing service account. Setting this value disables the automatic service account creation
-  ##
-  name: ""
```
[the remark this answers is not in the document]
Comment (FraPazGal) [reply bitnami/postgresql-ha/values.yaml:77]: I just moved to a different part of the `values.yaml`, under a different section, but GH can be a bit whiny about these types of things. Sorry for the confusion.

File: bitnami/postgresql-ha/templates/postgresql/service.yaml [hunk 5]
```diff
@@ -10,16 +11,22 @@ metadata:
     {{- if .Values.service.serviceLabels }}
     {{ toYaml .Values.postgresql.serviceLabels | indent 4 }}
     {{- end }}
```
[the remark this answers is not in the document]
Comment (FraPazGal) [reply bitnami/postgresql-ha/templates/postgresql/service.yaml]: You are right, this seems to be a typo. Changed to `.Values.service.serviceLabels`.

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml [hunk 6]
```diff
@@ -475,47 +459,43 @@ spec:
...
             httpGet:
               path: /
               port: metrics
-            initialDelaySeconds: {{ .Values.metrics.livenessProbe.initialDelaySeconds }}
-            periodSeconds: {{ .Values.metrics.livenessProbe.periodSeconds }}
-            timeoutSeconds: {{ .Values.metrics.livenessProbe.timeoutSeconds }}
-            successThreshold: {{ .Values.metrics.livenessProbe.successThreshold }}
-            failureThreshold: {{ .Values.metrics.livenessProbe.failureThreshold }}
-          {{- end }}
+          {{- else if .Values.metrics.customLivenessProbe }}
```
[the remark this answers is not in the document]
Comment (FraPazGal) [reply bitnami/postgresql-ha/templates/postgresql/statefulset.yaml:481]: Fixed, thanks!

File: bitnami/postgresql-ha/values.yaml (hunk 2 above)
Comment (fmulero) [reply bitnami/postgresql-ha/values.yaml]: 👍

**Review by fmulero — DISMISSED** [review]
LGTM

**Pull request merged**